friedybot.py

```diff
@@ -42,6 +43,126 @@ def __get_active_player_entries(self, player) -> PickupEntries:
             return PickupEntries.select().join(PickupGames).where(PickupGames.isPlayed == False).where(PickupEntries.playerId == player)
         return None
 
+    # this was basically taken from rcon2irc.pl
+    def __rgb_to_simple(self, r: int, g: int,b: int) -> int:
+
+        min_ = min(r,g,b)
+        max_ = max(r,g,b)
+
+        v = max_ / 15.0
+        s = (1 - min_/max_) if max_ != min_  else 0
+        if s < 0.2:
+            return 0 if v < 0.5 else 7
+
+        if max_ == min_:
+            h = 0
+        elif max_ == r:
+            h = (60 * (g - b) / (max_ - min_)) % 360
+        elif max_ == g:
+            h = (60 * (b - r) / (max_ - min_)) + 120
+        elif max_ == b:
+            h = (60 * (r - g) / (max_ - min_)) + 240
+
+        if h < 36:
+            return 1
+        elif h < 80:
+            return 3
+        elif h < 150:
+            return 2
+        elif h < 200:
+            return 5
+        elif h < 270:
+            return 4
+        elif h < 330:
+            return 6
+        else:
+            return 1
```

Comment:
maybe just my opinion, to make it a little more compact:
```suggestion
        if h < 36:
            return 1
        elif h < 80:
            return 3
        elif h < 150:
            return 2
        elif h < 200:
            return 5
        elif h < 270:
            return 4
        elif h < 330:
            return 6
        else:
            return 1
```
color_thresholds =[(36,1), (80,3), (150,2),(200,5),(270,4),(330,6)]

for theshold, value in color_thresholds:
    if h < threshold:
        return value

return 1